src/components/common/form/asyncValidate.js

```diff
@@ -0,0 +1,17 @@
+import axios from 'axios';
+
+const messageError = { email: 'That username is taken' };
+const emailExists = value => axios({
```

Comment:
I do not understand why this needs to be handled via redux action. This is an async function that I am importing in my register form and passing it to the redux form HOC so that it can check whether an email exists. 

https://redux-form.com/6.7.0/examples/asyncvalidation/

Reply:
Good question!

While the async validation is a convenient feature of redux-form, I believe it is an anti-pattern. The premise of redux is to manage the entire application state with reducers, action handlers and dispatchers. Action handlers define the actions you wish to perform, dispatchers trigger those actions and reducers update the state with the result. When you are directly passing an HTTP request method into the form, you are bypassing all these mechanisms. As a result, you will now have to start dealing with multiple ways for data passing through your application, which does not scale so well.

Quick example is.. when you need to pass in a cookie or auth token to the server for all authenticated requests, you'll have to update and manage multiple parts of your application instead of just one place (in the redux store).

Reply:
Is this actually possible? to check for async validation of my register form via "redux action". I gave it a try last night but i had no clue on how I would go to do this. I even tried to google the approach you suggested but everyone seemed to suggest doing it the way I did it. Could you please share some ideas.

Reply:
Ye it "should" be possible. I haven't actually implemented _field-level_ (I've only done form-level) async validation with redux-form. 

To do field-level async validation, you can try https://redux-form.com/8.1.0/examples/submitvalidation/. And use https://redux-form.com/8.1.0/docs/api/submissionerror.md/ to assign the error messages to specific fields.

Note: The validation in the above example will happen on submit (not on blur or on keydown).

Reply:
Hmm.. I was looking for a way to do it on blur/ keydown with some sort of redux action as you mentioned. I do not see any point having a validation on submit as the backend will throw an error if the user already exists to the client side. My goal was to trigger the validation before the user clicks the submit button

Reply:
I understand. So, there's a trade-off: anti-pattern for UX. If there's no other way to achieve it without an anti-pattern, then it's your call if it's worth it.

Reply:
I will create a new task to handle this in the future. I do not want to spend too much time on this feature which is not necessary for the MVP. For the time being, I will leave it as it is but I will not be adding more functionality that uses field validation (using this anti-pattern) to prevent a large refactoring in the future.

Reply:
Cool beans 👍